akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -32,7 +32,7 @@ private[akka] object RemoteWatcher {
   final case class UnwatchRemote(watchee: InternalActorRef, watcher: InternalActorRef)
 
   @SerialVersionUID(1L) case object Heartbeat extends HeartbeatMessage
-  @SerialVersionUID(1L) final case class HeartbeatRsp(addressUid: Int) extends HeartbeatMessage
+  @SerialVersionUID(1L) final case class HeartbeatRsp(addressUid: Long) extends HeartbeatMessage
```

Comment:
Those messages are only sent between systems that are not part of the same cluster. Within cluster there is other heartbeat mechanism.

Yeah, if we use java serialization it will be an interesting exercise to make that compatible.

If it's too difficult we could use new message types for these when Artery is enabled. We need to implement protobuf serialization for those anyway.